quixstreams/dataframe/windows/time_based.py

```diff
@@ -190,6 +203,13 @@ def wrapper(value: object) -> object:
             state_manager.get_store_transaction(store_name=store_name),
         )
         key = message_key()
+        if not key:
```

Comment:
Using `not` is incorrect here because it can filter out keys like `0` and `False`, which are valid values.

Use `is None` check instead